Chargebee/Classes/Authentication/CBAuthentication.swift

```diff
@@ -45,20 +45,26 @@ extension CBAuthenticationManager {
         guard key.isNotEmpty else {
             return onError(CBError.defaultSytemError(statusCode: 400, message: "SDK Key is empty"))
         }
-        let request = CBAPIRequest(resource: CBAuthenticationResource(key: key, bundleId: bundleId, appName: appName))
-        authenticateRestClient(network: request, logger: logger, handler: handler)
+        
+        if CBCache.shared.isCacheDataAvailable(key: key) {
+            CBCache.shared.readConfigDetails(key: key, logger: logger) { status in
```

Comment:
Can we use the `bundleId` as key instead? Since `key` is API key and since we are storing in UserDefaults, it will be readable.

Reply:
Do you need to pass the `logger` as well for the readConfig method?

Reply:
> Can we use the `bundleId` as key instead? Since `key` is API key and since we are storing in UserDefaults, it will be readable.

 @cb-haripriyan  If any other SDK tries to save something in **UserDefaults** with same key using `BundleID` [ `bundle_id` means AppID  have access to other SDKs and App, it might be used for some other reason right ? ] wouldn’t there be any issue ?
API key is specific to Chargebee SDK and looks unique there won’t be any problem right ?
Can you correct / Help me  on this  ?

Reply:
> Do you need to pass the `logger` as well for the readConfig method?


This  `buildResultHandlers(_ handler: @escaping CBResultHandler<T>, _ logger: CBLogger?)`
`buildResultHandlers` is expecting logger param otherwise it’s not working asking to enter logger value,So to create response handler  logger is required here though we don't have any use with it here.   `  let (onSuccess, _) = CBResult.buildResultHandlers(handler, logger)`

Reply:
We can use a combination of `bundleId + unique_identifier` instead.

Reply:
> We can use a combination of `bundleId + unique_identifier` instead.

sure